pulsar-client/src/main/java/org/apache/pulsar/client/impl/ProducerImpl.java

```diff
@@ -2214,8 +2233,8 @@ private void recoverProcessOpSendMsgFrom(ClientCnx cnx, MessageImpl from, long e
     /**
      *  Check if final message size for non-batch and non-chunked messages is larger than max message size.
      */
-    public boolean isMessageSizeExceeded(OpSendMsg op) {
-        if (op.msg != null && op.totalChunks <= 1) {
+    private boolean isMessageSizeExceeded(OpSendMsg op) {
+        if (op.msg != null && op.totalChunks <= 1 && !conf.isEncryptionEnabled()) {
```

Comment:
Disable size check if isChunkingEnabled?
```suggestion
        if (op.msg != null && !conf.isChunkingEnabled() && !conf.isEncryptionEnabled()) {
```

Reply:
Yes. And I will skip the check for `isEncryptionEnabled` here.